lib/logstash/inputs/imap.rb

```diff
@@ -63,14 +102,16 @@ def register
       # Ensure that the filepath exists before writing, since it's deeply nested.
       FileUtils::mkdir_p datapath
       @sincedb_path = File.join(datapath, ".sincedb_" + Digest::MD5.hexdigest("#{@user}_#{@host}_#{@port}_#{@folder}"))
+      @logger.debug? && @logger.debug("Generated sincedb path", sincedb_path: @sincedb_path)
     end
-    if File.directory?(@sincedb_path)
-      raise ArgumentError.new("The \"sincedb_path\" argument must point to a file, received a directory: \"#{@sincedb_path}\"")
-    end
-    @logger.info("Using \"sincedb_path\": \"#{@sincedb_path}\"")
```

Comment:
We lose a bit of information with this refactor.

If the path does not exist (from the perpective of the user running the Logstash process), and we later cannot write to it, we no longer have a record tying those details together unless the plugin is run with debug-level logging enabled.